desc/objectives/objective_funs.py

```diff
@@ -130,12 +170,53 @@ def build(self, use_jit=None, verbose=1):
             self._scalar = False
 
         self._set_derivatives()
+        sub_obj_jac_chunk_sizes_are_ints = [
+            isposint(obj._jac_chunk_size) for obj in self.objectives
+        ]
+        warnif(
+            any(sub_obj_jac_chunk_sizes_are_ints) and self._deriv_mode != "blocked",
+            UserWarning,
+            "'jac_chunk_size' was passed into one or more sub-objectives, but the"
+            " ObjectiveFunction is  using 'batched' deriv_mode, so sub-objective "
+            "'jac_chunk_size' will be ignored in favor of the ObjectiveFunction's "
+            f"'jac_chunk_size' of {self._jac_chunk_size}."
+            " Specify 'blocked' deriv_mode if each sub-objective is desired to have a "
+            "different 'jac_chunk_size' for its Jacobian computation.",
+        )
+        warnif(
+            self._jac_chunk_size not in ["auto", None]
+            and self._deriv_mode == "blocked",
+            UserWarning,
+            "'jac_chunk_size' was passed into ObjectiveFunction, but the "
+            "ObjectiveFunction is using 'blocked' deriv_mode, so sub-objective "
+            "'jac_chunk_size' are used to compute each sub-objective's Jacobian, "
+            "`ignoring the ObjectiveFunction's 'jac_chunk_size'.",
+        )
+
         if not self.use_jit:
             self._unjit()
 
         self._set_things()
-
         self._built = True
+
+        if self._jac_chunk_size == "auto":
+            # Heuristic estimates of fwd mode Jacobian memory usage,
+            # slightly conservative, based on using ForceBalance as the objective
+            estimated_memory_usage = 2.4e-7 * self.dim_f * self.dim_x + 1  # in GB
+            max_chunk_size = round(
+                (desc_config.get("avail_mem") / estimated_memory_usage - 0.22)
```

Comment:
some notes about `desc_config.get("avail_mem")`:
- In all cases, it's the available memory when desc was first imported, which may be different than the amount available for objective functions/optimization.
- For CPU on clusters, it returns the total memory on the node, not just the amount that was requested as part of a slurm job, so usually its a massive over-estimate which might cause issues. I'm not sure what the best option is, I tried a while back to make it give the actual allocated memory but it seems like that value depends on a few different slurm options so it's not straightforward to parse

Reply:
Hm yea I knew about the second option, I feel like rarely on CPU do we do intensive stuff on the cluster.

I can add a disclaimer in the ObjectiveFunction docstring about this issue on the cluster if on a CPU